### PR TITLE
[800] CI workflow for updating tedge reference docs

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes

To update `tedge cli` reference docs the [script](`https://github.com/thin-edge/thin-edge.io/tree/main/docs/gen-ref-docs.sh` ) has to be run manually and create the PR manually.
This PR automates this process by adding a GH action workflow to run this script on the `1st of every month`. 
Also, this workflow can be triggered manually. 
It creates the PR automatically and sends the PR review request.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
May I ask why it is necessary to have the `<command> --help` output in the docs? If a user knows that a command can be called with `--help`, they receive the exact same output as the docs show... so where is the benefit of having of also having the `--help`-output in the docs? :thinking:

**Comment by didier-wenzek** [comment]
> May I ask why it is necessary to have the <command> --help output in the docs?

Just to have this online. Nothing more than what we can have `man` online. Nothing fancy, I agree. But cheap.

File: .github/workflows/update-tedge-ref-docs.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          uses: actions/checkout@v2
+        - run: |
+           git config --global user.email "info@thin-edge.io"
+           git config --global user.name "Versioneer"
+           
+        - name: enable toolchain via github action
+          # https://github.com/actions-rs/toolchain
+          uses: actions-rs/toolchain@v1
+          with:
+            toolchain: 1.58
```
Comment (abelikt) [inline .github/workflows/update-tedge-ref-docs.yml:25]: Lets also use 1.58.1 here

**Comment by abelikt** [comment]
I'm fine with it as long itsyitsy is the monthly reviewer :)

**Review by abelikt — APPROVED** [review]
Approved with one small update request

**Pull request merged**